### ISSUE TITLE
Customize grant flows per application: config doorkeeper grant_flows passing a block.

### ISSUE THREAD

**Issue by phlegx** [doorkeeper-gem/doorkeeper#1245]
Hi!

Can I pass a block to:

```ruby
Doorkeeper.configure do
  ...
  grant_flows %w[authorization_code client_credentials]
end
```

like in `custom_access_token_expires_in`:

```ruby
Doorkeeper.configure do
  ...
  custom_access_token_expires_in do |context|
    context.client.some_method
  end
end
```

Passing a block to `grant_flows` can help setting up different grant flows per application. Example:

```ruby
Doorkeeper.configure do
  ...
  grant_flows do |client|
    client.available_grant_flows
  end
end
```

**Comment by nbulaj** [comment]
Hi @phlegx . It's impossible at the moment. Also this configuration option is designed for the entire server and knows nothing about aoolications (clients). It defines grant flows supported by your server

**Comment by phlegx** [comment]
Hi @nbulaj. But can be added as feature? 

Yes, the server is one, but in Doorkeeper I can add many applications. Why should every application have all the default grant flows? Single application, single configuration, multiple applications, multiple configurations. I mean, why there are context blocks for `custom_access_token_expires_in` and `use_refresh_token`? Because they are useful to have better config granularity. The same could be very useful for `grant_flows`.

**Comment by phlegx** [comment]
Found issue #1207 with the same feature request!

**Comment by nbulaj** [comment]
Landed in `master` with #1286
